lib/traject_config.rb

```diff
@@ -128,6 +134,10 @@
 to_field 'pub_created_display', extract_marc('260abcefg:264abcefg')
 to_field 'pub_created_s', extract_marc('260abcefg:264abcefg', :first => true)
 
+to_field 'pub_citation_display' do |record, accumulator|
```

Comment:
Since the set_pub_citation method returns an array, the accumulator array should be replaced with the return value of the function:
```ruby
to_field 'pub_citation_display' do |record, accumulator|
  pub_info = set_pub_citation(record)
  accumulator.replace(pub_info)
end
```